Add workaround for Scope Mismatch issue for HttpClientFactory (#34864)
dotnet/docs: docs/core/extensions/httpclient-factory.md

1. Create an instance of `TodoService`, passing in the instance of `HttpClient` to its constructor.

> [!IMPORTANT]

> > Using typed clients in singleton services can be dangerous. For more information, see the [Using Typed clients in singleton services](#using-typed-clients-in-singleton-services) section.

Author: Rephrased to "Avoid" because the section is about dangers and problems of such approach

Users are strongly advised **not to cache scope-related information** (such as data from `HttpContext`) inside `HttpMessageHandler` instances and use scoped dependencies with caution to avoid leaking sensitive information.

If you require access to an application DI scope from your message handler, e.g. for authentication, you can work around that by incapsulating scope-aware logic in a separate transient `DelegatingHandler`, and wrap it around `HttpMessageHandler` instance from the `IHttpClientFactory` cache, which you can get using `IHttpMessageHandlerFactory` by calling <xref:System.Net.Http.IHttpMessageHandlerFactory.CreateHandler%2A> for any registered _named client_. In that case, you would need to create `HttpClient` instance yourself using the constructed handler.

> :::image type="content" source="media/httpclientfactory-scopes-workaround.png" alt-text="Diagram showing gaining access to application DI scopes via a separate transient message handler and IHttpMessageHandlerFactory":::

Reviewer: ```suggestion
:::image type="content" source="media/httpclientfactory-scopes-workaround.png" alt-text="Diagram showing gaining access to app DI scopes via a separate transient message handler and IHttpMessageHandlerFactory":::
```

> The following example shows creating an `HttpClient` with a scope-aware `HttpHandler`:

Reviewer: ```suggestion
The following example shows creating an `HttpClient` with a scope-aware `HttpMessageHandler`:
```

:::code source="snippets/http/scopeworkaround/ScopeAwareHttpClientFactory.cs" id="CreateClient":::

> A further workaround can follow with an extension method for registering a scope-aware `HttpHandler` and overriding default `IHttpClientFactory` registration by a transient service with access to the current application scope:

Reviewer: ```suggestion
A further workaround can follow with an extension method for registering a scope-aware `HttpMessageHandler` and overriding default `IHttpClientFactory` registration by a transient service with access to the current app scope:
```
